runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -254,10 +257,12 @@ protected boolean fillReadQueue(final long maxGlobal,
                         }
                     }
                 }
-
-                // If we STILL don't have the data, now we need to do a hole
-                // fill.
-                if (currentEntry.getType() == DataType.EMPTY) {
+                //FIXME To be replaced by cleaner interface. This is just a patch till the real fix comes in.
+                // If we STILL don't have the data, it means the writer is either dead or too slow.
+                // We will now try to make sure we have committed data for the currentEntry. We will first
+                // try a HOLE FILL, failing which we will re-read the entry to see if the writer completed.
+                // We will try this in a loop till we get committed data for the currentEntry.
+                while (currentEntry.getType() == DataType.EMPTY) {
```

Comment:
Is this patch really needed? Why not just fix the underlying problem (bug i reported #439). Also, this fix only works for a single node. Please consider a chain where the head is not the tail. For example, client1 writes to the head, but doesn't complete its write to the tail(it dies), client2 reads the tail and finds null. Consequently, client2 tries to fill a whole, but can't since it will encounter an overwrite exception when it tries to write to the head. Now client2 will wait forever for another client to propagate the write, but it should try to propagate that value itself as part of the fill hole mechanism. 

The hole filling mechanism, should not only write holes, but should propagate partial writes.